salt/utils/find.py

```diff
@@ -655,10 +659,22 @@ def find(self, path):
                             if result is not None:
                                 yield result
 
-            if depth == self.maxdepth:
+            if self.maxdepth is not None and depth > self.maxdepth:
                 dirs[:] = []
 
 
+def path_depth(path):
+    depth = 0
+    head = path
+    while True:
```

Comment:
This looks like something that can infinitely recurse. If we don't hit the break on the first iteration, what causes this not to loop endlessly?